Apache benchmarks made dynamic

abishekchiffon opened the pull request:
## Description

Made the following changes in the apache benchmarks to make it more dynamic.
1. Config.properties can be dynamically set on run
2. inference_model_url can be dynamically set on run
3. inference_url and ports are made to be taken dynamically from the config.properties

This builds on top of the work of @fbbradheintz 

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

- Logs
[gpu_mnist_explanations_kf.log](https://github.com/pytorch/serve/files/5696726/gpu_mnist_explanations_kf.log)
[gpu_mnist_explnations_ts.log](https://github.com/pytorch/serve/files/5696727/gpu_mnist_explnations_ts.log)
[gpu_mnist_predictions_kf.log](https://github.com/pytorch/serve/files/5696728/gpu_mnist_predictions_kf.log)
[gpu_mnist_predictions_ts.log](https://github.com/pytorch/serve/files/5696729/gpu_mnist_predictions_ts.log)
[gpu_resnet_predictions.log](https://github.com/pytorch/serve/files/5696730/gpu_resnet_predictions.log)



## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

[docker_benchmark.log](https://github.com/pytorch/serve/files/5710558/docker_benchmark.log)
[config_json_benchmark.log](https://github.com/pytorch/serve/files/5710605/config_json_benchmark.log)
[config_json.log](https://github.com/pytorch/serve/files/5710606/config.log)

harshbafna reviewed (changes requested):
@abishekchiffon Please address the following:

 - Update documentation to run benchmarking with KFServing APIs
 - Add steps to run this with docker container
 - Attach log for run with cofig.json file with new parameters
 - Attach logs for runs on docker container

abishekchiffon commented:
> @abishekchiffon Please address the following:
> 
> * Update documentation to run benchmarking with KFServing APIs
> * Add steps to run this with docker container
> * Attach log for run with cofig.json file with new parameters
> * Attach logs for runs on docker container

@harshbafna 
* Updated documentation to run benchmarking with KFServing APIs
* Added steps to run this with docker container
* Attached log for run with cofig.json file with new parameters
* Attached logs for runs on docker container
* The apache bench docker worked after building from local.
